aiqleads/core/project_tracking.py

```
@@ -125,19 +220,43 @@
+            
+        # Check for directory traversal
+        if '..' in normalized_path:
+            logger.error(f"Directory traversal detected: {normalized_path}")
             return False
             
-        # TODO: Add more thorough validation against project_structure
+        # Ensure path matches project structure
+        allowed_dirs = ['core', 'utils', 'scripts', 'data', 'tests']
+        path_parts = Path(path).parts
+        if len(path_parts) > 1:
+            if path_parts[1] not in allowed_dirs:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)